[SERV-7498] Reboots Demo from Scratch as a cleaner, quicker solution to debugging the old demo

AmitaiB opened the pull request:
## This PR simply replaces the old iOS SDK demo project with a ground-up rewrite that serves the same (a) customer-facing and (b) Support, issue-reproduction purposes.

#### This is because debugging and fixing the old demo would require far more engineering and support resources than the 20 minutes it took to re-create this demo from scratch.

IOW, there _are_ many "changes", but this PR is intended to be discontinuous as far as the codebase is concerned since it meets all the acceptance criteria of the old demo.

**Background Details**
1. A customer reported that -- only when bitcode was enabled -- project archiving for release would fail with an IPA tool error.
  1. Support reproduced this in the (old) demo app, but also got the error _with our SDK removed entirely_.
  2. Support escalated [SERV-7498](https://jwplayer.atlassian.net/browse/SERV-7498) to fix the demo app.
  3. The Mobile L2 hypothesized that the issue was some obscure project setting.
  4. To test that hypothesis, Support wrote this demo from scratch; this new demo indeed does not exhibit the error.
  4. Support closed the ticket, and now uses this demo instead.

[SERV-7498]: https://jwplayer.atlassian.net/browse/SERV-7498?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

AmitaiB committed:
Reboots Demo in Xcode 11.6

henryhlee commented on Podfile:
```diff
@@ -1,10 +1,10 @@
 
   # Pods for JWPlayer-SDK-iOS-Demo
-	pod 'JWPlayer-SDK'
+  pod 'JWPlayer-SDK', '3.16.2'
```
shouldn't the version string be '~> 3.16.2' so it will always find the latest iOS SDK?

AmitaiB replied:
In this instance, I don't think so.

`~>` is great for dependency management to model a rule like "the latest version that will not break anything", but it turns out that this project is mostly used as an Issue Reproduction app, which means switching back and forth to compare _specific_ versions is the more common use case. The version the customer wants is running is rarely the latest. 

It also lets us instruct less technical customers to just change the version number string to the one they/we want.

henryhlee commented on Podfile.lock:
```diff
@@ -1,16 +1,16 @@
 PODS:
-  - JWPlayer-SDK (3.13.0)
+  - JWPlayer-SDK (3.16.2)
```
I don't think this file should be checked in at all. This is generated dynamically by the cocopods.

AmitaiB replied:
Added `Podfile.lock` to the gitignore.

AmitaiB replied:
I also made sure `Pods/` was not commented-out in the gitignore.

henryhlee reviewed (dismissed):
The JWPlayer_iOS_SDK.framework should not be part of the git depot, it is supposed to be coming from the CocoaPods. Plus, Podfile.lock should not in the git as well.

MikeSalvador reviewed (dismissed):
Seems good to me

s-seibert reviewed (dismissed):
You should remove the .DS_Store file. It should not be included. Check .gitignore to see if it is in there.

AmitaiB committed:
Incorporates reviewer feedback (1)

AmitaiB commented:
@henryhlee - I incorporated your suggestion to add the `Podfile.lock` to the gitignore.
@s-seibert  - I incorporated your suggestion, restoring the `.DS_Store` to the gitignore.

Henry, I note that I no longer have write-access to this repo, so I cannot merge the PR even after following the reviews.

david-almaguer commented:
@AmitaiB I think this project will only run on iOS 13, should it run on previous iOS versions?

AmitaiB committed:
Updates to SDK 3.18.1

AmitaiB commented:
This repo is unused. Closing the PR.
